src/main/java/de/tum/in/www1/artemis/service/CourseScoreCalculationService.java

```diff
@@ -98,35 +124,154 @@ public CourseScoresDTO calculateCourseScores(long courseId, Collection<Long> stu
+            PlagiarismVerdict mostSeverePlagiarismVerdict = null;
+            boolean hasParticipated;
+            PlagiarismMapping plagiarismMapping = PlagiarismMapping.createFromPlagiarismCases(plagiarismCases);
+            if (entry.getValue().isEmpty()) {
+                presentationScorePassed = false;
+                hasParticipated = false;
+            }
+            else if (plagiarismMapping.studentHasVerdict(entry.getKey(), PlagiarismVerdict.PLAGIARISM)) {
+                presentationScorePassed = false;
+                mostSeverePlagiarismVerdict = PlagiarismVerdict.PLAGIARISM;
+                hasParticipated = true;
+            }
+            else {
+                presentationScorePassed = isPresentationScoreSufficientForBonus(studentScores.getPresentationScore(), course.getPresentationScore());
+                Map<Long, PlagiarismCase> plagiarismCasesForStudent = plagiarismMapping.getPlagiarismCasesForStudent(entry.getKey());
+                mostSeverePlagiarismVerdict = findMostServerePlagiarismVerdict(plagiarismCasesForStudent.values());
+                hasParticipated = true;
+            }
+            studentScoresForExamBonusSource
+                    .add(new StudentScoresForExamBonusSourceDTO(studentScores.getAbsoluteScore(), studentScores.getRelativeScore(), studentScores.getCurrentRelativeScore(),
+                            studentScores.getPresentationScore(), entry.getKey(), presentationScorePassed, mostSeverePlagiarismVerdict, hasParticipated));
+        });
+
+        return new CourseScoresForExamBonusSourceDTO(maxAndReachablePoints.maxPoints, maxAndReachablePoints.reachablePoints, course.getPresentationScore(),
+                studentScoresForExamBonusSource);
+    }
+
+    /**
+     * Get all the items needed for the CourseForDashboardDTO.
+     * This includes scoresPerExerciseType and participationResults.
+     *
+     * @param course the course to calculate the items for.
+     * @param userId the id of the students whose scores in the course will be calculated.
+     * @return the CourseForDashboardDTO containing all the mentioned items.
+     */
+    public CourseForDashboardDTO getScoresAndParticipationResults(Course course, long userId) {
+        List<StudentParticipation> studentParticipations = new ArrayList<>();
+        for (Exercise exercise : course.getExercises()) {
+            exercise.setCourse(course);
+            StudentParticipation exerciseParticipation;
+            // This method is used in the CourseResource where the course is first fetched with lazy participations, and participations are then fetched separately and added to the
+            // course if found.
+            // If no participations are found for the course, no value is set to the course's participations and trying to access them here would throw a
+            // LazyInitializationException.
+            if (Hibernate.isInitialized(exercise.getStudentParticipations())) {
```

Comment:
Shouldn't we invert this logic? If it is not yet initialized, load it from the database?

Reply:
This is due to how the CourseService adds participations the course's exercises (`CourseService: fetchParticipationsWithSubmissionsAndResultsForCourses`). First, the exercises are extracted from the courses, participations are then retrieved from the database, and then for each exercise the student participations are only set, if there were fetched any for that exercise in the first place (`ExerciseService: filterForCourseDashboard`). The result is, that for each exercise that does not have any participations, a `LazyInitializationException` would be thrown if trying to access those participations. Or how would you implement that?

Reply:
I meant something like 'if(!Hibernate.isInitialized...) exercise.setStudentParticipations(studentParticipationRepository.find(...))'

Reply:
I ran the test and as expected, I never got participations at that point when using `if (!Hibernate.isInitialized(exercise.getStudentParticipations()) {studentParticipationRepository.findByExerciseIdAndStudentId()}`. This makes sense because participations are already fetched in the `CourseService`: `studentParticipationRepository.getAllParticipationsOfUserInExercises(user, exercises);` It's just that they are only actually set to the exercise, if any came back from that database call. So trying to find them in the here again is unnecessary. I tried to clarify things in the comment.

Reply:
Can we change the logic in `getAllParticipationsOfUserInExercises`? So that the participations are always initialized, but are an empty list then? I don't like interpreting hibernate exceptions as a valid status with some meaning. Especially since this method is public, one could also pass exercises which are not loaded this way and would break that assumption.